docs/example_go.md

```diff
@@ -3,29 +3,303 @@ title: "Example: Go"
+
+# Records the current time, then kicks off a server update.
+# Normally, you would let Tilt do deploys automatically, but this
+# shows you how to set up a custom workflow that measures it.
+local_resource(
+    'deploy',
+    './record-start-time.sh',
+)
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second
+argument is the script that it runs.
+
+We've also modified our server to read that start time, calculate the time
+elapsed, then display this in both logs and HTML.
+
+Let's click the button on the `deploy` resource and see what happens!
 
-## Servantes
+<figure>
+  <a class="is-image" href="https://cloud.tilt.dev/snapshot/AbakyOYLWWvW6J2BJFo=">
+    <img src="assets/docimg/example-go-image-2.png">
+  </a>
+  <figcaption>Step 1 complete. Click the screenshot to see an interactive snapshot.</figcaption>
+</figure>
 
-[Servantes](https://github.com/windmilleng/servantes) is a personalized homepage
-app implemented with many microservices in multiple languages. You can run it
-with the commands
+| Approach | Deploy Time |
+|---|---|
+| Naive | 4.2s |
+{:.benchmark-report}
 
+If you look closely, the elapsed time displayed in the Tilt sidebar is different
+than the benchmark our app logged. That's OK! In multi-service development,
+there are many benchmarks we care about -- the time to build the image, the time
+to schedule the process, and the time until the server is ready to serve
+traffic. 
+
+The Tilt sidebar gives you some default benchmarks, and the tools to capture
+your own benchmarks.
+
+Our benchmarks show this is slow. Can we do better?
+
+## Step 2: Let's Optimize for the Go Compiler
+
+What's taking up so much time? The logs show that when we make the change to a file, we:
+
+1) Copy the source files to the image.
+2) Download the `gorilla/mux` dependency.
+3) Re-compile the Go binary from scratch.
+
+But the Go team has done a lot of work to make caching dependencies and
+incremental compiles fast.  How can we better use the Go tools how they're meant
+to be used?
+
+With `local_resource`, we can compile the Go binary locally, and copy the binary to
+the container.
+
+Here's our [new Tiltfile](https://github.com/windmilleng/tilt-example-go/blob/master/2-optimized/Tiltfile) 
+with the following new code:
+
+```python
+local_resource(
+  'example-go-compile',
+  'CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o build/tilt-example-go ./',
+  deps=['./main.go', './start.go'],
+  resource_deps = ['deploy'])
+  
+docker_build(
+  'example-go-image',
+  '.',
+  dockerfile='deployments/Dockerfile',
+  only=[
+    './build',
+    './web',
+  ])
 ```
-git clone https://github.com/windmilleng/servantes
-cd servantes; tilt up
+
+We've added a `local_resource()` that compiles the Go binary locally for our
+Linux container.
+
+We've also added a new `only` parameter to `docker_build()`.  We don't need the
+source files in the Docker build context. We only need the `./build` directory
```

Comment:
```suggestion
source files in the Docker build context; we only need the `./build` directory
```

Reply:
I think you're right that we "disagree on the reason" that the `only` parameter needs explanation. Yes only pulling in the directories we need makes thing more efficient, but I'd rather talk explicitly about how this affects file-watching because that's the place I foresee people getting stuck.

Like, I might look at this and say "yeah it's a little more efficient to pull in only what we need, but doesn't seem like a huge deal, what happens if i leave this step out?" and then I see things double-building or building out of order; i'd be totally confused and there's nothing in the doc to help me understand what's happening.